src/providers/hana/qgshanadataitemguiprovider.cpp

```diff
@@ -0,0 +1,171 @@
+#include "qgshanaprovider.h"
+#include "qgshanautils.h"
+#include "qgsnewnamedialog.h"
+
+#include <QInputDialog>
+#include <QMessageBox>
+
+void QgsHanaDataItemGuiProvider::populateContextMenu(
+  QgsDataItem *item, QMenu *menu, const QList<QgsDataItem *> &, QgsDataItemGuiContext )
+{
+  if ( QgsHanaRootItem *rootItem = qobject_cast<QgsHanaRootItem *>( item ) )
+  {
+    QAction *actionNew = new QAction( tr( "New Connection…" ), this );
+    connect( actionNew, &QAction::triggered, this, [rootItem] { newConnection( rootItem ); } );
+    menu->addAction( actionNew );
+  }
+
+  if ( QgsHanaConnectionItem *connItem = qobject_cast<QgsHanaConnectionItem *>( item ) )
+  {
+    QAction *actionRefresh = new QAction( tr( "Refresh" ), this );
+    connect( actionRefresh, &QAction::triggered, this, [connItem] { refreshConnection( connItem ); } );
+    menu->addAction( actionRefresh );
+
+    menu->addSeparator();
+
+    QAction *actionEdit = new QAction( tr( "Edit Connection…" ), this );
+    connect( actionEdit, &QAction::triggered, this, [connItem] { editConnection( connItem ); } );
+    menu->addAction( actionEdit );
+
+    QAction *actionDelete = new QAction( tr( "Delete Connection" ), this );
+    connect( actionDelete, &QAction::triggered, this, [connItem] { deleteConnection( connItem ); } );
+    menu->addAction( actionDelete );
+  }
+
+  if ( QgsHanaSchemaItem *schemaItem = qobject_cast<QgsHanaSchemaItem *>( item ) )
+  {
+    QAction *actionRefresh = new QAction( tr( "Refresh" ), this );
+    connect( actionRefresh, &QAction::triggered, this, [schemaItem] { schemaItem->refresh(); } );
+    menu->addAction( actionRefresh );
+  }
+}
+
+bool QgsHanaDataItemGuiProvider::deleteLayer( QgsLayerItem *item, QgsDataItemGuiContext )
+{
+  if ( QgsHanaLayerItem *layerItem = qobject_cast<QgsHanaLayerItem *>( item ) )
+  {
+    const QgsHanaLayerProperty &layerInfo = layerItem->layerInfo();
+    QString objectName = QStringLiteral( "%1.%2" ).arg( layerInfo.schemaName, layerInfo.tableName );
+    QString typeName = layerInfo.isView ? tr( "View" ) : tr( "Table" );
+
+    if ( QMessageBox::question( nullptr, tr( "Delete %1" ).arg( typeName ),
+                                tr( "Are you sure you want to delete %1?" ).arg( objectName ),
+                                QMessageBox::Yes | QMessageBox::No, QMessageBox::No ) != QMessageBox::Yes )
+      return true;
+
+    QgsHanaConnectionItem *connItem = qobject_cast<QgsHanaConnectionItem *>( layerItem->parent()->parent() );
+    QgsHanaConnectionRef conn( connItem->name() );
+    if ( conn.isNull() )
+      return false;
+
+    QString sql = QStringLiteral( "DROP TABLE %1.%2" )
+                  .arg( QgsHanaUtils::quotedIdentifier( layerInfo.schemaName ), QgsHanaUtils::quotedIdentifier( layerInfo.tableName ) );
+
+    QString errMessage;
+    bool res = conn->execute( sql, &errMessage );
+    if ( !res )
+    {
+      QMessageBox::warning( nullptr, tr( "Delete %1" ).arg( typeName ), errMessage );
```

Comment:
Please change all occurrences of `::warning` and `::information` to `notify()`, using this API: https://github.com/qgis/QGIS/blob/master/src/gui/qgsdataitemguiprovider.h#L185

Reply:
Done in 47934888cb.